pkg/volume/portworx/portworx_util.go

```diff
@@ -182,6 +186,55 @@ func (util *PortworxVolumeUtil) UnmountVolume(u *portworxVolumeUnmounter, mountP
 	return nil
 }
 
+func (util *PortworxVolumeUtil) ResizeVolume(spec *volume.Spec, newSize resource.Quantity, volumeHost volume.VolumeHost) error {
+	driver, err := util.getPortworxDriver(volumeHost, false /*localOnly*/)
+	if err != nil || driver == nil {
+		glog.Errorf("Failed to get portworx driver. Err: %v", err)
+		return err
+	}
+
+	vols, err := driver.Inspect([]string{spec.Name()})
+	if err != nil {
+		return err
+	}
+
+	if len(vols) != 1 {
+		return fmt.Errorf("failed to inspect Portworx volume: %s. Found: %d volumes", spec.Name(), len(vols))
+	}
+
+	vol := vols[0]
+	newSizeInBytes := uint64(volutil.RoundUpToGiB(newSize) * volutil.GIB)
+	if vol.Spec.Size >= newSizeInBytes {
+		glog.Infof("Portworx volume: %s already at size: %d greater than or equal to new "+
+			"requested size: %d. Skipping resize.", vol.Spec.Size, newSizeInBytes)
+		return nil
+	}
+
+	vol.Spec.Size = newSizeInBytes
+	err = driver.Set(spec.Name(), vol.Locator, vol.Spec)
+	if err != nil {
+		return err
+	}
+
+	// check if the volume's size actually got updated
+	vols, err = driver.Inspect([]string{spec.Name()})
+	if err != nil {
+		return err
+	}
+
+	if len(vols) != 1 {
+		return fmt.Errorf("failed to inspect resized Portworx volume: %s. Found: %d volumes", spec.Name(), len(vols))
+	}
+
+	updatedVol := vols[0]
+	if updatedVol.Spec.Size != vol.Spec.Size {
```

Comment:
I don't think we need to check exact equality, may be just `updatedVol.Size >= vol.Size`  should be sufficient?

Reply:
Done. Now I'm allowing the updated volume to be greater than the requested size.